### ISSUE TITLE
docker-compose.with_db.yml healthchecks fail out of the box

### ISSUE THREAD

**Issue by Brutus5000** [railslove/ebicsbox#278]
The postgres database in `docker-compose.with_db.yml` fails to startup by default.

The error for that is that the `POSTGRES_USER` is not available for the healtcheck via pg_isready, because compose does not lookup the environment variables of a service. Unless it is specifically declared in an .env file it fails to run. And since everything depends on the postgres db, the whole stack fails to launch.

https://github.com/railslove/ebicsbox/blob/6ca2cecbf5a079c6621f99ec9e7c83ac6a8cb760/docker-compose.with_db.yml#L15-L46

Potential solution: move the default environment to .env.example and mention to copy the file to .env in the readme.

**Comment by janz93** [comment]
Hi @Brutus5000,

Thank you for reporting this issue. I want to start by apologizing for the delay in addressing your report—this seems to have slipped through the cracks, and I missed it for quite some time.  

I’ve tested the `docker-compose.with_db.yml` configuration on my machine, and everything appears to be working as expected. The PostgreSQL database starts up correctly, and the health check via `pg_isready` does not fail in my setup.  

To help me better understand and resolve the issue, could you provide more detailed information about your environment and setup? Specifically:  

1. **Operating System and Docker/Docker Compose Versions**  
   - What OS are you using?  
   - What are the exact versions of Docker and Docker Compose?  

2. **Modifications to `docker-compose.with_db.yml`**  
   - Have you made any changes to the file? If so, could you share the modifications?  

3. **Environment Variables**  
   - Are you using an `.env` file? If so, could you share the relevant portions? *(Please ensure you exclude any sensitive information.)*  

4. **Steps to Reproduce and Logs**  
   - Could you describe the exact steps to reproduce the issue?  
   - Could you provide the full error logs you’re seeing?  

I look forward to your response so we can figure out what might be causing this issue. Thanks for your patience!

**Comment by Brutus5000** [comment]
Hello there,

quite a while, so I had to make sure the problem still exists for the current version.

My setup:
* MacOs 14.7
* Docker engine 27.3.1
* Docker Compose version v2.29.7-desktop.1


Steps to reproduce (on a blank git clone):
```sh
cp .env.example .env
docker compose -f docker-compose.with_db.yml up -d db
docker compose -f docker-compose.with_db.yml ps
```

The status will move to unhealthy. This is due to the fact tha the `$POSTGRES_USER` in line 27 cannot be resolved. It is only declared in the service environment and not in the .env.example. So it is not available for resolving inside the compose file. 

https://github.com/railslove/ebicsbox/blob/6ca2cecbf5a079c6621f99ec9e7c83ac6a8cb760/docker-compose.with_db.yml#L27

As a result the healthcheck fails after around a minute.

I solved this by appending this to my local `.env` file:

```
POSTGRES_DB=ebicsbox
POSTGRES_USER=ebicsbox
POSTGRES_PASSWORD=password
```
